### ISSUE TITLE
Build broken in 3.3.0 in conf_lib.c on AIX 7.1

### ISSUE THREAD

**Issue by Eason-Yu** [openssl/openssl#24149]
Got the following build error on AIX 7.1 ppc even if patching the fixes of https://github.com/openssl/openssl/commit/491bbb444c4b654de14adc7031eb44e88a101edb

```
gcc  -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include  -maix64 -pthread -ggdb -O0 -maix64 -ggdb -O0 -maix64 -maix64 -DB_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/share/centrifydc/ssl\"" -DENGINESDIR="\"/usr/share/centrifydc/lib64/engines-3\"" -DMODULESDIR="\"/usr/share/centrifydc/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG -DPURIFY -I/usr/share/centrifydc/include -I/usr/share/centrifydc/kerberos/include -MMD -MF crypto/conf/libcrypto-lib-conf_lib.d.tmp -MT crypto/conf/libcrypto-lib-conf_lib.o -c -o crypto/conf/libcrypto-lib-conf_lib.o crypto/conf/conf_lib.c
In file included from /usr/include/sys/limits.h:42:0,
                 from /usr/include/limits.h:30,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.1.0.0/4.6.3/include-fixed/limits.h:169,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.1.0.0/4.6.3/include-fixed/syslimits.h:7,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.1.0.0/4.6.3/include-fixed/limits.h:34,
                 from include/internal/e_os.h:13,
                 from crypto/conf/conf_lib.c:12:
/usr/include/standards.h:92:28: error: operator '==' has no left operand
make[1]: *** [crypto/conf/libcrypto-lib-conf_lib.o] Error 1
```

**Comment by rsbeckerca** [comment]
I think this has been fixed by #24083

**Comment by Eason-Yu** [comment]
> I think this has been fixed by #24083

I applied the change of #24083, but still got this build error. FYI, I also tried to build 3.1.2 on the same machine, works well.

**Comment by Eason-Yu** [comment]
@rsbeckerca I made a typo while patching the fixes. This ticket should be false alarm. Sorry about it.
